corehq/apps/formplayer_api/smsforms/tests/test_formplayer_interface.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+import requests_mock
+from django.conf import settings
+from django.test import TestCase
+
+from corehq.apps.formplayer_api.smsforms.api import FormplayerInterface
+from corehq.apps.formplayer_api.utils import get_formplayer_url
+from corehq.util.hmac_request import get_hmac_digest
+
+
+class TestFormplayerInterface(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.session_id = uuid.uuid4().hex
+        self.domain = uuid.uuid4().hex
+        self.user_id = uuid.uuid4().hex
+        self.interface = FormplayerInterface(self.session_id, self.domain, self.user_id)
+
+    def test_get_raw_instance(self):
+        expected_request_data = {
+            'action': 'get-instance',
+            'session-id': self.session_id,
+            'session_id': self.session_id,
+            'domain': self.domain,
+            'oneQuestionPerScreen': True,
+            'nav_mode': 'prompt',
+        }
+        self._test_request("get-instance", self.interface.get_raw_instance, expected_request_data, {})
+
+    def test_answer_question(self):
+        mock_response = {"event": {
+            "datatype": "select",
+            "choices": ["red", "green", "blue"],
+            "caption": "What's your favorite color?",
+            "type": "question",
+            "answer": None,
+            "required": 0,
+            "ix": "1",
+            "help": None
+        }}
```

Comment:
What is the value of this? I'm assuming this was meant as a test to document what we expect to send and receive, but only the sending portion is enforced. It seems like we could return `{}` here as well, and the test would still function. If we aren't doing anything with the return value (like parsing it to return a domain-friendly object), this seems like the type of thing that would be better left in a comment on the function definition

Reply:
We have to return a valid response otherwise the code interacting with the API will raise an exception. We don't necessarily need ALL elements of the response. This was largely copied from the [XformsEvent docstring](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/formplayer_api/smsforms/api.py#L85).